pocs/sensors/arduino_io.py

```diff
@@ -0,0 +1,209 @@
+        port, parse it and find a 'name' attribute in the top-level object.
+        Else returns None.
+    """
+    if not logger:
+        logger = get_root_logger()
+    logger.debug('Attempting to connect to serial port: {}'.format(port))
+    serial_reader = None
+    try:
+        # First open a connection to the device.
+        try:
+            serial_reader = open_serial_device(port)
+            if not serial_reader.is_connected:
+                serial_reader.connect()
+            logger.debug('Connected to {}', port)
+        except Exception as e:
+            logger.warning('Could not connect to port: {}'.format(port))
+            return None
+        try:
+            reading = serial_reader.get_and_parse_reading(retry_limit=3)
+            if not reading:
+                return None
+            (ts, data) = reading
+            if isinstance(data, dict) and 'name' in data and isinstance(data['name'], str):
+                return data['name']
+            logger.warning('Unable to find board name in reading: {!r}', reading)
+            return None
+        except Exception as e:
+            logger.error('Exception while auto-detecting port {!r}: {!r}'.format(port, e))
+    finally:
+        if serial_reader:
+            serial_reader.disconnect()
+
+
+def open_serial_device(port, serial_config=None, **kwargs):
+    # Using a long timeout (2 times the report interval) rather than retries which can just break
+    # a JSON line into two unparseable fragments.
+    sd_kwargs = dict(baudrate=9600, retry_limit=1, retry_delay=0, timeout=4.0, name=port)
+    if serial_config:
+        sd_kwargs.update(serial_config)
+    sd_kwargs.update(kwargs)
+    sd_kwargs['port'] = port
+    return rs232.SerialData(**sd_kwargs)
+
+
+class ArduinoIO(object):
+    """Reads the output from an arduino, and exposes the relays for change."""
+
+    def __init__(self, board_name, port, output_queue, serial_config=None):
+        """Inits for board on device.
+
+        Args:
+            board_name:
+                The name of the board, used as the name of the database
+                table/collection to write to.
+            port:
+                The device to connect to.
+            output_queue:
+                The queue to which to send the readings. Will not block
+                on sending (put call), so the queue should have room for
+                a few entries. By not blocking on put, we avoid dropping
+                records while reading from the serial line due to dropped
+                chars, and thus corrupted JSON.
+            serial_config:
+                The portion of the config file descibing the serial settings.
+        """
+        self.board_name = board_name or port
+        self.port = port
+        self._serial_data = open_serial_device(
+            port, serial_config=serial_config, name=board_name)
+        self._output_queue = output_queue
+        self._logger = get_root_logger()
```

Comment:
I thought about using PanBase, but it calls config.load_config in such a way that it loads the pocs config files, rather than enabling the caller to provide the config or to provide the config file names.

Reply:
Hmm, I was thinking we passed the `*args, **kwargs` into the underlying `load_config`, which is what we should do. Then you could pass `conf_files` to any constructor and it will load just those files.

Reply:
Let's do that in another PR.